File: invoice-service/src/main/java/com/mcb/billing/invoice/messaging/InvoiceIssuedEvent.java
package com.mcb.billing.invoice.messaging;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * Event published when an invoice is issued. {@code eventId} is a fresh UUID per
 * publish so consumers can process it idempotently (deduplicate redeliveries).
 */
public record InvoiceIssuedEvent(
        String eventId,
        String invoiceNumber,
        String customerNumber,
        BigDecimal totalAmount,
        String currency,
        LocalDate dueDate
) {
}

File: notification-service/src/main/kotlin/com/mcb/billing/notification/messaging/InvoiceIssuedEvent.kt
package com.mcb.billing.notification.messaging

import java.math.BigDecimal
import java.time.LocalDate

/**
 * The notification-service's own view of the InvoiceIssued event. Each service
 * defines its own copy of the contract (no shared code); unknown JSON fields are
 * ignored. {@code eventId} drives idempotent processing.
 */
data class InvoiceIssuedEvent(
    val eventId: String,
    val invoiceNumber: String,
    val customerNumber: String,
    val totalAmount: BigDecimal,
    val currency: String,
    val dueDate: LocalDate,
)
